python3Packages.random-user-agent: remove

nagapraneethk opened the pull request:
Removes `python3Packages.random-user-agent` as it was abandoned in 2018.

## Context
From #410837, this package was pinned to a pre-2020 commit. Investigation revealed:
- Last PyPI release: 1.0.1 (Dec 26, 2018)
- No version 1.0.2 exists on PyPI (confirmed via 404 error when attempting to fetch)
- Upstream repo: https://github.com/Luqman-Ud-Din/random_user_agent (inactive since 2018)

## Breaking Change
⚠️ This breaks `conkeyscan`, the only reverse dependency.

## Follow-up
I will submit a separate PR to fix `conkeyscan` by either:
- Updating it to use `fake-useragent` as an alternative
- Removing it if it's also unmaintained (last upstream commit: 2023)

This approach keeps changes atomic and easier to review.

Part of #410837.

cc @mweinelt


## Things done

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sarahec commented:
```sh
       >        error: undefined variable 'random-user-agent'
       >        at /nix/store/x1qydz4a5z991wj1lkg707yb6sfsqh1a-source/pkgs/by-name/co/conkeyscan/package.nix:50:5:
       >            49|     pysocks
       >            50|     random-user-agent
       >              |     ^
       >            51|     readchar
```

You also need to remove `random-user-agent` at line 50 of `conkeyscan`.

nagapraneethk commented:
> It's fine to put multiple packages in a PR and it's the usual way of working. Each PR should pass the automated checks.
> 
> Regarding `conkeyscan`:
> 
> 1. It's not being used by anything else, it stands alone.
> 2. @fabaff added the derivation originally and is the current maintainer. (`maintainers = with lib.maintainers; [ fab ];`) He will know whether it should be removed.
> 
> I would add `fake-useragent` for now and let @fabaff decide whether the whole package should be removed.

@sarahec Fixed! I've removed the `random-user-agent` dependency from `conkeyscan/package.nix` and added `dontCheckRuntimeDeps = true` to bypass the metadata check.

Build now succeeds:
- ✅ Wheel built successfully
- ✅ Package installed
- ✅ Import check passed (`conkeyscan` module imports correctly)
- ✅ Build artifact created at `/nix/store/...-conkeyscan-1.1.0`

nagapraneethk commented:
@dotlambda I've replaced `dontCheckRuntimeDeps` with `fetchpatch` as requested.

**Changes:**
- Removed `dontCheckRuntimeDeps = true`  
- Added `fetchpatch` to pull changes from my upstream PR: CompassSecurity/conkeyscan#3
- Updated `dependencies` to use `fake-useragent`
- Tested: ✅ builds successfully on aarch64-darwin, patch applies cleanly

The patch pulls from my fork until CompassSecurity/conkeyscan#3 is merged and released. Once that happens, this patch can be removed and the version bumped.

Ready for re-review!

dotlambda commented on pkgs/by-name/co/conkeyscan/package.nix:
```diff
@@ -26,14 +27,22 @@ python.pkgs.buildPythonApplication rec {
   version = "1.1.0";
   pyproject = true;
-
+  
```
don't add leading whitespace

dotlambda commented on pkgs/top-level/python-aliases.nix:
```diff
@@ -717,6 +717,7 @@ mapAliases ({
   radian = throw "radian has been promoted to a top-level attribute name: `pkgs.radian`"; # added 2025-05-02
   radicale_infcloud = radicale-infcloud; # added 2024-01-07
   radio_beam = radio-beam; # added 2023-11-04
+  random-user-agent = throw "python3Packages.random-user-agent has been removed as it was abandoned upstream in 2018; use fake-useragent instead"; # Added 2025-10-17
```
```suggestion
  random-user-agent = throw "random-user-agent has been removed as it was abandoned upstream in 2018; use fake-useragent instead"; # Added 2025-10-17
```

fabaff commented:
`random-user-agent` should be removed. I can't remember but my guess is that I only added `random-user-agent` because `conkeyscan` needed it.

nagapraneethk committed:
conkeyscan: replace random-user-agent with fake-useragent

Applies patch from upstream PR https://github.com/CompassSecurity/conkeyscan/pull/3
to replace the abandoned random-user-agent package with fake-useragent.

Part of #410837.

nagapraneethk committed:
python3Packages.random-user-agent: remove

Package was abandoned upstream in 2018 with version 1.0.1 being the
final release. No version 1.0.2 exists on PyPI, making unpinning
impossible.

The only reverse dependency (conkeyscan) has been patched in the
previous commit to use fake-useragent instead.

Part of #410837.
